tools/deployment/package-helm/templates/compression-worker-deployment.yaml

```diff
@@ -0,0 +1,108 @@
+                do echo Waiting for queue to be ready...;
+                sleep 2;
+              done
+        - name: "wait-for-redis"
+          image: "busybox:latest"
+          command:
+            - "sh"
+            - "-c"
+            - >-
+              until nc -z {{ include "clp.fullname" . }}-redis 6379;
+                do echo Waiting for redis to be ready...;
+                sleep 2;
+              done
+      containers:
+        - name: "compression-worker"
+          image: "{{ include "clp.image.ref" . }}"
+          imagePullPolicy: "{{ .Values.image.clpPackage.pullPolicy }}"
+          env:
+            - name: "BROKER_URL"
+              value: "amqp://{{ .Values.credentials.queue.username }}:{{ .Values.credentials.queue.password }}@{{ include "clp.fullname" . }}-queue:5672"
+            - name: "CLP_CONFIG_PATH"
+              value: "/etc/clp-config.yaml"
+            - name: "CLP_HOME"
+              value: "/opt/clp"
+            - name: "CLP_LOGGING_LEVEL"
+              value: {{ .Values.clpConfig.compression_worker.logging_level | quote }}
+            - name: "CLP_LOGS_DIR"
+              value: "/var/log/compression_worker"
+            - name: "CLP_WORKER_LOG_PATH"
+              value: "/var/log/compression_worker/worker.log"
+            - name: "PYTHONPATH"
+              value: "/opt/clp/lib/python3/site-packages"
+            - name: "RESULT_BACKEND"
+              value: "redis://default:{{ .Values.credentials.redis.password }}@{{ include "clp.fullname" . }}-redis:6379/{{ .Values.clpConfig.redis.compression_backend_database }}"
+          volumeMounts:
+            - name: "config"
+              mountPath: "/etc/clp-config.yaml"
+              subPath: "clp-config.yaml"
+              readOnly: true
+            - name: "compression-worker-logs"
+              mountPath: "/var/log/compression_worker"
+            - name: "archives"
+              mountPath: "/var/data/archives"
+            - name: "logs-input"
+              mountPath: "/mnt/logs"
+              readOnly: true
+            - name: "tmp"
+              mountPath: "/var/tmp"
+          command: [
+            "python3", "-u",
+            "/opt/clp/lib/python3/site-packages/bin/celery",
+            "-A", "job_orchestration.executor.compress",
+            "worker",
+            "--concurrency", "{{ .Values.workerConcurrency }}",
+            "--loglevel", "WARNING",
+            "-f", "/var/log/compression_worker/worker.log",
+            "-Q", "compression",
+            "-n", "compression-worker"
+          ]
+      volumes:
+        - name: "config"
+          configMap:
+            name: {{ include "clp.fullname" . }}-config
+        - name: "compression-worker-logs"
+          persistentVolumeClaim:
+            claimName: {{ include "clp.fullname" . }}-compression-worker-logs
+        - name: "archives"
+          persistentVolumeClaim:
+            claimName: {{ include "clp.fullname" . }}-archives
```

Comment:
Do you think it's useful to deduplicate things like these volumes since we're repeating them in different deployments? That would avoid cases where we have a typo in "archives" or something.